src/main.ts

```diff
@@ -1090,27 +1022,28 @@ function action_input_bool(name: string): boolean {
 }
 
 async function main(): Promise<void> {
-  try {
-    let correlation: string = actions_core.getState("correlation");
-    if (correlation === "") {
-      correlation = `GH-${randomUUID()}`;
-      actions_core.saveState("correlation", correlation);
-    }
-
-    const installer = new NixInstallerAction(correlation);
+  const installer = new NixInstallerAction();
 
+  try {
     const isPost = actions_core.getState("isPost");
+    actions_core.saveState("isPost", "true");
     if (isPost !== "true") {
       await installer.detectAndForceDockerShim();
       await installer.install();
-      actions_core.saveState("isPost", "true");
     } else {
       await installer.cleanupDockerShim();
       await installer.report_overall();
     }
   } catch (error) {
     if (error instanceof Error) actions_core.setFailed(error);
   }
+
+  await installer.idslib.complete();
```

Comment:
It *looks* like this is "finally"-ish enough to fire in case anything during install crashes, but maybe even just move it down to 1049, in an explicit `finally`?

Reply:
I've cleaned this up a lot, by delegating the actual execution of this code as callbacks that the IdsToolbox calls.